src/components/description_list/index.d.ts

```diff
@@ -12,6 +12,8 @@ declare module '@elastic/eui' {
     compressed?: boolean;
     textStyle?: EuiDescriptionListTextStyle;
     type?: EuiDescriptionListType;
+    titleProps?: object;
```

Comment:
To better constraint the available properties, use `HTMLAttributes<HTMLElement>` instead of `object`

```suggestion
    titleProps?: HTMLAttributes<HTMLElement>;
```